tests/tracer/test_service.py

```diff
@@ -25,3 +28,209 @@ def _start(self):
     with pytest.raises(RuntimeError):
         s.start()
     assert s.status == service.ServiceStatus.STOPPED
+
+
+def test_service_copy():
+    s = service.Service()
+    assert s.copy() == s.copy()
+    assert s.copy() is not s.copy()
+    s.start()
+    s2 = s.copy()
+    assert s is not s2
+    assert s == s2
+    assert s.status == service.ServiceStatus.RUNNING
+    assert s2.status == service.ServiceStatus.STOPPED
+    s.stop()
+    assert s is not s2
+    assert s == s2
+    assert s.status == service.ServiceStatus.STOPPED
+    assert s2.status == service.ServiceStatus.STOPPED
+
+
+def test_service_manager():
+    class MyService(service.Service):
+        pass
+
+    class MyServiceProcess(service.ServiceProcess):
+        SERVICE_CLASS = MyService
+
+    sm = MyServiceProcess()
+    sm.start()
+    assert sm.status == sm._service.status == service.ServiceStatus.RUNNING
+
+    sm.stop()
+    assert sm.status == sm._service.status == service.ServiceStatus.STOPPED
+
+
+def call_program(*args):
```

Comment:
Seems like we should have this as a test util if we don't already 🙂

Reply:
I thought about it but every test suite out there has its own version with its own use/edge case. I was not excited by the idea to try to write a one-solve-them-all function for all of those.

Reply:
fair enough :+1: